### PR TITLE
LPD-20019

### PR DESCRIPTION
**Pull request by icgsbr**
https://liferay.atlassian.net/browse/LPD-20019

### CODE REVIEW

**Review by carolmariaabb — CHANGES_REQUESTED** [review]
@icgsbr please also remove from [here](https://github.com/liferay-forms/liferay-portal/blob/9efb53ba93ba039765650f46647446531d5adcb0/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMContentLocalServiceImpl.java#L32). 

Make sure to run `buildService` again.

**Comment by icgsbr** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMContentLocalServiceImpl.java [hunk 1]
```diff
@@ -22,14 +22,11 @@
...
  * @author     Brian Wing Shun Chan
  * @author     Eduardo Lundgren
- * @deprecated As of Cavanaugh (7.4.x), replaced by {@link
- *             DDMFieldLocalServiceImpl}
  */
 @Component(
 	property = "model.class.name=com.liferay.dynamic.data.mapping.model.DDMContent",
 	service = AopService.class
 )
-@Deprecated
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMContentLocalServiceImpl.java:32]: @icgsbr please also run `buildService` after this commit, I believe the `DDMContentLocalServiceBaseImpl` will be updated.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml [hunk 2]
```diff
@@ -35,10 +35,6 @@
 		<finder name="CompanyId" return-type="Collection">
 			<finder-column name="companyId" />
 		</finder>
-
-		<!-- References -->
-
-		<reference entity="User" package-path="com.liferay.portal" />
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml:41]: @icgsbr why did you remove this? what is the purpose of having this? please make sure nothing will break.

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMContentLocalServiceImpl.java (hunk 1 above)
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMContentLocalServiceImpl.java:32]: done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml (hunk 2 above)
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml:41]: The source format was failing with the following message: 

Avoid using reference 'User' for Entity 'DDMContent', use private variables in *ServiceImpl instead: /home/igorcosta/dev/projects/liferay-portal/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml (SourceCheck:XMLServiceReferenceCheck)

        at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:458)
        at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:291)

Could you help me on make this work? @carolmariaabb

Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml:41]: This was the only way I managed to make it pass the test and by testing, I saw no errors. Yet, I'm almost sure there's another way of doing it, but idk it.

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml:41]: @icgsbr first all of, you need to run `buildService` after removing this (always run `buildSevice` after changing the `service.xml` file). 
It's fine to remove it, but you'll probably need to update the `DDMContentLocalServiceImpl` by adding the private variable as suggested:

```	
@Reference
private UserLocalService _userLocalService;
``` 

I believe this [line](https://github.com/liferay/liferay-portal/blob/918d2a92fea02807d92eec05a2d4ab8903e11aea/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/base/DDMContentLocalServiceBaseImpl.java#L599-L600) will be removed from `DDMContentLocalServiceBaseImpl` after running `buildService`, so this suggested change will fix the compilation error [here](https://github.com/liferay/liferay-portal/blob/918d2a92fea02807d92eec05a2d4ab8903e11aea/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMContentLocalServiceImpl.java#L41).

Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml:41]: I also managed to see the use of the User on DDMContentLocalServiceImpl and it is defined inside a method, not being a class variable

**Comment by carolmariaabb** [comment]
@icgsbr this code is not compiling, please fix it:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml (hunk 2 above)
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/service.xml:41]: Done

**Comment by icgsbr** [comment]
Hi, @carolmariaabb!

One of the CSE's told me the client wants this fix delivered by today due to their go-live; therefore, if anything needs to be altered, please tell me so I can fix it in order to start doing the hotfix ASAP.

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:report:2549074824

**Comment by marcelabc** [comment]
ci:report:2549067044

**Comment by marcelabc** [comment]
After analyzing the CI results for the ci:test:relevant, I found this failure that may be unique to this PR:

- com.liferay.journal.notifications.test.JournalUserNotificationTest

**Comment by carolmariaabb** [comment]
Thanks, @marcelabc!

1. [com.liferay.journal.notifications.test.JournalUserNotificationTest](https://testray.liferay.com/home/-/testray/case_results/2549075957?tab=history) is a flaky.  It has failed with the same error 5 times before.
2. [LocalFile.VirtualInstancesUpgrade#ViewVirtualInstancesArchive7413U33](https://testray.liferay.com/home/-/testray/case_results/2549069427?tab=history). I would say it's related since we have the failure after `Reindexing of com.liferay.dynamic.data.mapping.model.DDMTemplate entities`

image

@icgsbr please take a look

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward
